### DISCUSSION TITLE
DLRS calculating incorrectly

### DISCUSSION THREAD

**Discussion by DanWLG** [SFDO-Community/declarative-lookup-rollup-summaries#1267] [Q&A]
Hi All,

I am using the DLRS tool ([provider](https://andyinthecloud.com))

We are experiencing an issue where the lookup rollup summaries are not calculating correctly or not at all and to get it to work we have to click the "calculate" button manually to run the calculations.

What could be going wrong here?

**Comment by aheber** [comment]
@DanWLG, you will get better support through our [Trailblazer Community group](https://success.salesforce.com/_ui/core/chatter/groups/GroupProfilePage?g=0F9300000009O5p), most of our troubleshooting and other expert functions are handled there.

I'll do my best to help here. You haven't provided any details about your setup so I don't have anywhere to begin troubleshooting. 

Do you have your rollups set to "Active?" Do you have many rollups or just a few? Do some work or none work? Does the rollup work sometimes and not others? What "Mode" do you have these rollups set to? Are they Realtime, Scheduled, Process Builder, or Developer?

I know we've seen some that set to "Scheduled" but forget to schedule the job that processes the pending rollup items. Could this be your case?

Hopefully by answering those questions we can start to get some idea about how your system is configured and what might need to change.

**Reply by stephenbrcnz** [reply]
Also in developer mode, even if you have installed the trigger on the relevant object, it won't run automatically - it's designed to be called from other code. If you are seeing correct results when using the calc button but not automatically on child record update then possibly you're using developer or scheduled mode but haven't set up the relevant schedule or call.

**Reply by DanWLG** [reply]
@aheber 

We have about 40 rollups and they are all mainly real time or scheduled and 1 process builder. Some of the roll ups work and some don't. The roll ups that should work are all active.

The scheduled ones look to be scheduled but this is my first time dealing with DLRS so I could be mistaken.

@stephenbrcnz 
When you mentioned developer mode, do you mean the "calculation mode"?

**Reply by stephenbrown1** [reply]
@DanWLG yes the calc mode = developer, some more info here: https://sfdo-community-sprints.github.io/DLRS-Documentation/Architecture/calculates.html#important-note
(although it prob needs fleshing out a bit more about developer mode... something that I'll add to the doc backlog :)
